src/sagemaker_training/mpi.py

```diff
@@ -95,18 +95,20 @@ def run(
 
         if wait:
             logger.info("Waiting for MPI process to finish.")
-            _wait_orted_process_to_finish()
+            self._wait_master_to_finish()
             logger.info("Orted process exited")
-            time.sleep(30)
         logger.info("MPI process finished.")
 
     def _wait_master_to_start(self):  # type: () -> None
         while not _can_connect(self._master_hostname):
             time.sleep(1)
 
     def _wait_master_to_finish(self):  # type: () -> None
+        logger.info("Check master node connectivity")
+        sleep_timer = 30
```

Comment:
why we set 30s here? could this time be reduced since the costumer would be billed for extra seconds.

Reply:
That's a trade off we have to choose. We can't have high timer because that would bill our cx, we can't have low timer as it would unnecessarily establish frequent `ssh` connections.

Reply:
agreed, lets keep 30s for now.